src/doc/src/reference/source-replacement.md

```diff
@@ -72,12 +78,16 @@ replace-with = "my-vendor-source"
 replace-with = "another-source"
 
 # Several kinds of sources can be specified (described in more detail below):
-registry = "https://example.com/path/to/index"
+registry = "ssh://git@example.com/path/to/index.git"
+registry = "sparse+https://example.com/path/to/index"
 local-registry = "path/to/registry"
 directory = "path/to/vendor"
 
-# Git sources can optionally specify a branch/tag/rev as well
-git = "https://example.com/path/to/repo"
+# Git sources are only used to identify git-based dependencies
+# to replace when vendoring
+[source."git+https://example.com/project/crate"]
```

Comment:
While this doesn't fit within the literal intent of the document (see https://github.com/rust-lang/cargo/pull/15974/files#r2353426174), I think it can fit within the spirit of it but we should probably update the comment for `the-source-name` to clarify that that is for registry sources